plugin.py

```diff
@@ -69,112 +76,29 @@ def check_binary(cls) -> bool:
-            sublime.error_message("Error downloading XML server binary!")
-
-        cls.thread = None
-
-    @classmethod
-    def setup(cls) -> None:
-        if cls.thread or cls.ready:
-            return
-
-        # read server source information
-        filename = "Packages/{}/server.json".format(__package__)
-        server_json = sublime.decode_value(sublime.load_resource(filename))
-
-        cls.version = server_json["version"]
-        cls.url = sublime.expand_variables(server_json["url"], {"version": cls.version})
-        cls.checksum = server_json["sha256"].lower()
-
-        # built local server binary path
-        dest_path = package_cache()
-        cls.binary = os.path.join(dest_path, os.path.basename(cls.url))
-
-        # download server binary on demand
-        cls.ready = cls.check_binary()
-        if not cls.ready:
-            cls.thread = threading.Thread(target=cls.download)
-            cls.thread.start()
-
-        # clear old server binaries
-        for fn in os.listdir(dest_path):
-            fp = os.path.join(dest_path, fn)
-            if fn[-4:].lower() == ".jar" and not os.path.samefile(fp, cls.binary):
-                try:
-                    os.remove(fp)
-                except OSError:
-                    pass
-
-    @classmethod
-    def cleanup(cls) -> None:
-        cls.binary = None
-        cls.checksum = None
-        cls.ready = False
-        cls.url = None
-        cls.version = None
-
-
-class LspXMLPlugin(LanguageHandler):
-
-    def __init__(self):
-        super().__init__()
-        LspXMLServer.setup()
-
-    @property
-    def name(self) -> str:
-        return __package__.lower()
-
-    @property
-    def config(self) -> ClientConfig:
-        settings_file = "LSP-lemminx.sublime-settings"
-        
-        client_config = {
-            "enabled": True,
-            "command": ["java", "-jar", LspXMLServer.binary],
-        }
-
-        default_config = sublime.decode_value(
-            sublime.load_resource(
-                "Packages/{}/{}".format(__package__, settings_file)
-            )
-        )
-
-        user_config = sublime.load_settings(settings_file)
-        for key, value in merge_configs(default_config, user_config):
-            client_config[key] = value
-
-        return read_client_config(self.name, client_config)
-
-    def on_start(self, window) -> bool:
-        if not is_java_installed():
-            sublime.status_message(
-                "Please install Java Runtime for the XML language server to work."
-            )
-            return False
-        if not LspXMLServer.ready:
-            sublime.status_message("Language server binary not yet downloaded.")
-            return False
-        return True
-
 
 def plugin_loaded() -> None:
-    LspXMLServer.setup()
+    # read server source information
+    filename = "Packages/{}/server.json".format(__package__)
+    server_json = sublime.decode_value(sublime.load_resource(filename))
```

Comment:
Unrelated to the change but would it make sense to have the content of server.json just be a constant in `plugin.py`? I feel that this is an unnecessary complication.

Reply:
1. The server.json is basically the counter part of what package.json means for the npm language servers. 
2. Data should be kept separate from code
3. We could also ask why all the schema files in e.g. LSP-json are not part of the plugin.py as well. The server.json doesn't contain as much info, but the meaning is just the same.